### PR TITLE
Lower-level primitives for working with subscriptions

### PR DESCRIPTION
**Pull request by fsoikin**
Some lower-level primitives to support less trivial subscription-related code in https://github.com/collegevine/app/pull/18834

### CODE REVIEW

File: src/Elmish/Subscription.purs [hunk 1]
```diff
@@ -113,7 +117,49 @@ subscribe f = subscribeMaybe (Just <<< f)
...
+quasiBind :: ∀ m a b. (a -> Aff b) -> Subscription m a -> Subscription m b
+quasiBind f (Subscription go) =
+  Subscription \dispatch ->
+    go \a -> launchAff_ do
+      b <- f a
+      liftEffect $ dispatch b
+
+-- | Given a `Maybe`-producing subscription, creates a new subscription that
+-- | filters out `Nothing` values and fires only on receiving `Just`.
+hush :: ∀ m a. Subscription m (Maybe a) -> Subscription m a
```
Comment (gasi) [inline src/Elmish/Subscription.purs:162]: FYI, the existing `hush` is `Either a b -> Maybe b`: https://pursuit.purescript.org/packages/purescript-either/6.1.0/docs/Data.Either#v:hush

How about `compact` (from Ruby) or something similar?

**Review by gasi — APPROVED** [review]
LGTM 👍 

TIL: You added subscriptions to Elmish

File: src/Elmish/Subscription.purs (hunk 1 above)
Comment (fsoikin) [reply src/Elmish/Subscription.purs:162]: Yes, I know the existing `hush`, that's why I named it this way. I don't envision this function being used in most regular consumer code, only in special cases, and in those cases I intend it to be used qualified as in `Subscription.hush`.

Comment (gasi) [reply src/Elmish/Subscription.purs:162]: What about `catMaybes` as it nicely follows the existing Signature of `Array (Maybe a) -> Array a`?

https://pursuit.purescript.org/packages/purescript-arrays/7.3.0/docs/Data.Array#v:catMaybes

Comment (mcordova47) [reply src/Elmish/Subscription.purs:162]: FYI, late to the party, but looks like there is actually a [Compactable](https://pursuit.purescript.org/packages/purescript-filterable/5.0.0/docs/Data.Compactable#t:Compactable) typeclass that has exactly this shape:

```purs
class Compactable f where
  compact :: forall a.
    f (Maybe a) -> f a
```

Comment (gasi) [reply src/Elmish/Subscription.purs:162]: @mcordova47 Nice find! 

> To be compactable alone, no laws must be satisfied other than the type signature.

😂

**Pull request merged**